File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章拼数.py
#拼数lanqiaoOJ 782

n=int(input())
nums = input().split()
for i in range(0,n-1):
    for j in range(i+1,n):
        if nums[j]+nums[i] > nums[i]+nums[j]:   
           nums[j],nums[i] = nums[i],nums[j]    #交换两个数的位置
print(''.join(nums ))

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章拼数.java
//拼数 lanqiaoOJ 782

import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] nums = new int[n];
        for(int i = 0; i < n; i++)
            nums[i] = sc.nextInt();
        String[] s = new String[n];
        String ans = "";
        for(int i = 0; i < n; i++)
            s[i] = nums[i] + "";
        for(int i = 0; i < n - 1; i++)  //交换排序
            for(int j = i + 1; j < n; j++) 
                if((s[j] + s[i]).compareTo(s[i] + s[j]) < 0) {
                    String temp = s[j];
                    s[j] = s[i];
                    s[i] = temp;
                }
        for(int i = n - 1; i >= 0; i--) 
            ans += s[i];
        System.out.println(ans);
    }
}
